### PR TITLE
Add Tractyl Manuform 6x7

### PR DESCRIPTION
**Pull request by matthew-dews**
Adds a Tractyl Manuform 6x7 with support for Elite C controller

## Description

This code was based on the existing code for tractyl_manuform/5x6_right/elite_c

Also adds some #error statements to flag issues when building.
While adding this keyboard, I ran into an issue that would have been caught
if there was #error in tractyl_manuform.h

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Just a heads up, i have some upcoming changes to the base code for tractyl manuform (to bring it in line with the charybdis, as that better handling for a bunch of stuff)

Otherwise, looks good, and good catch.

**Comment by matthew-dews** [comment]
Thanks for the heads up, I'll keep an eye out for it

**Comment by matthew-dews** [comment]
I have a separate keymap that I made for myself. I think it's better than the default one, which I based off the dactyl manuform 6x7, which I think I based off of some other default.

Can I include that separate keymap in this (or another) PR? I'm not sure what the policy is in regards to having user keymaps in the main qmk_firmware repo.

**Comment by drashna** [comment]
If it's for the same keyboard that is being PRed, then that is fine.  but if it's a different keyboard, then ideally, a separate pr is ideal.

**Comment by matthew-dews** [comment]
Re-pushed the commit with an additional keymap

**Comment by matthew-dews** [comment]
Hi Nick, I was just able to build this an hour ago. Can you include the command you are using to build it and the error output?

**Comment by tzarc** [comment]
> Hi Nick, I was just able to build this an hour ago. Can you include the command you are using to build it and the error output?

Apologies for the lack of clarity; looks like it might be a question for @drashna -- it works against `master`, but not against `develop`.

As for the command:
```sh
qmk compile -kb handwired/tractyl_manuform/6x7_right/elite_c -km all
```

**Comment by drashna** [comment]
> > Hi Nick, I was just able to build this an hour ago. Can you include the command you are using to build it and the error output?
> 
> Apologies for the lack of clarity; looks like it might be a question for @drashna -- it works against `master`, but not against `develop`. Invariably, if/when this gets merged to `master` it'll automatically get merged to `develop` and start failing.

correct.  the tm_sync code right now syncs both the pointing device reports, as well as the dpi config.  Both of which are handled by core code on develop.  Which means that tm_sync.c has been removed on develop. 

I'm fine with fixing this upstream, when it's merged.  Or (ideally, as it's more simple), remove the "src += tm_sync.c" stuff, and compile from develop until develop is merged back.

**Comment by matthew-dews** [comment]
I see that develop has been merged to master recently. Congrats on that work.

I've made some changes to this code recently and should be making another pass in a week or so. Will look at getting this PR in shape to merge again

File: keyboards/handwired/tractyl_manuform/6x7_right/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+AUDIO_ENABLE = no           # Audio output
+SWAP_HANDS_ENABLE = no
+
+POINTING_DEVICE_ENABLE = yes
+POINTING_DEVICE_DRIVER = pmw3360
+MOUSE_SHARED_EP = yes
+
+SPLIT_KEYBOARD = yes
+
+QUANTUM_LIB_SRC += tm_sync.c
```
Comment (drashna) [inline keyboards/handwired/tractyl_manuform/6x7_right/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
Honestly, I think you only need one change to get this working.

**Pull request closed without merging**